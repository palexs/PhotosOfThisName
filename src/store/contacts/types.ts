import {Contact} from 'react-native-contacts';
import {
  LOAD_CONTACTS_START,
  LOAD_CONTACTS_SUCCESS,
  LOAD_CONTACTS_FAILURE,
  SET_PERMISSIONS_GRANTED,
} from './constants';

export interface ContactsState {
  loading: boolean;
  contacts: Contact[];
  permissionsGranted: boolean | null;
  error: Error | null;
}

interface LoadContactsStart {
  type: typeof LOAD_CONTACTS_START;
}

interface LoadContactsSuccess {
  type: typeof LOAD_CONTACTS_SUCCESS;
  contacts: Contact[];
}

interface LoadContactsFailure {
  type: typeof LOAD_CONTACTS_FAILURE;
  error: Error;
}

interface SetPermissionsGranted {
  type: typeof SET_PERMISSIONS_GRANTED;
  granted: boolean;
}

export type Action =
  | LoadContactsStart
  | LoadContactsSuccess
  | LoadContactsFailure
  | SetPermissionsGranted;
