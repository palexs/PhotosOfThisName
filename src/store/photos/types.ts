import {ActionCreator} from 'redux';
import {ThunkAction} from 'redux-thunk';
import {Dependencies, RootState} from '../types';
import {
  SEARCH_START,
  SEARCH_SUCCESS,
  SEARCH_FAILURE,
  LOAD_MORE_START,
  LOAD_MORE_SUCCESS,
  LOAD_MORE_FAILURE,
  GET_LOCATION_START,
  GET_LOCATION_SUCCESS,
  GET_LOCATION_FAILURE,
  TRACK_OPEN_URL,
  TRACK_OPEN_URL_FAILURE,
} from './constants';

type NumericBool = 0 | 1;

export interface Photo {
  id: string;
  owner: string;
  secret: string;
  server: string;
  farm: number;
  title: string;
  ispublic: NumericBool;
  isfriend: NumericBool;
  isfamily: NumericBool;
}

export interface PhotosResponse {
  page: number;
  pages: number;
  perpage: number;
  total: number;
  photo: Photo[];
}

export interface PhotoLocationResponse {
  id: string;
  location: {
    latitude: string;
    longitude: string;
    accuracy: string;
    context: string;
    locality: {
      _content: string;
    };
    county: {
      _content: string;
    };
    region: {
      _content: string;
    };
    country: {
      _content: string;
    };
    neighbourhood: {
      _content: string;
    };
  };
}

export enum PhotoSize {
  thumbnail75 = 's',
  thumbnail100 = 't',
  thumbnail150 = 'q',
  small240 = 'm',
  small320 = 'n',
  small400 = 'w',
  medium800 = 'c',
  large1024 = 'b',
  large1600 = 'h',
  large2048 = 'k',
}

export interface PhotoLocation {
  loading: boolean;
  country: string | null;
  error: Error | null;
}

export interface PhotosState {
  query: string;
  fetching: boolean;
  loadingMore: boolean;
  data: Photo[];
  error: Error | null;
  totalPages: number;
  locations:
    | {
        [photoID: string]: PhotoLocation;
      }
    | {};
}

export interface SearchStart {
  type: typeof SEARCH_START;
  name: string;
}

export interface SearchSuccess {
  type: typeof SEARCH_SUCCESS;
  totalPages: number;
  photos: Photo[];
}

export interface SearchFailure {
  type: typeof SEARCH_FAILURE;
  error: Error;
}

export interface LoadMoreStart {
  type: typeof LOAD_MORE_START;
}

export interface LoadMoreSuccess {
  type: typeof LOAD_MORE_SUCCESS;
  photos: Photo[];
}

export interface LoadMoreFailure {
  type: typeof LOAD_MORE_FAILURE;
  error: Error;
}

export interface GetLocationStart {
  type: typeof GET_LOCATION_START;
  photoID: string;
}

export interface GetLocationSuccess {
  type: typeof GET_LOCATION_SUCCESS;
  country: string;
  photoID: string;
}

export interface GetLocationFailure {
  type: typeof GET_LOCATION_FAILURE;
  error: Error;
  photoID: string;
}

export interface TrackOpenURL {
  type: typeof TRACK_OPEN_URL;
}

export interface TrackOpenURLFailure {
  type: typeof TRACK_OPEN_URL_FAILURE;
  error: Error;
}

export type Action =
  | SearchStart
  | SearchSuccess
  | SearchFailure
  | LoadMoreStart
  | LoadMoreSuccess
  | LoadMoreFailure
  | GetLocationStart
  | GetLocationSuccess
  | GetLocationFailure
  | TrackOpenURL
  | TrackOpenURLFailure;

export type ThunkActionCreator = ActionCreator<
  ThunkAction<Promise<void>, RootState, Dependencies, Action>
>;
