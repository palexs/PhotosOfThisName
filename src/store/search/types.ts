import {
  SEARCH_START,
  SEARCH_SUCCESS,
  SEARCH_FAILURE,
  LOAD_MORE_START,
  LOAD_MORE_SUCCESS,
  LOAD_MORE_FAILURE,
  TRACK_OPEN_URL,
} from './constants';

type NumericBool = 0 | 1;

export interface Photo {
  id: string;
  owner: string;
  secret: string;
  server: string;
  farm: number;
  title: string;
  ispublic: NumericBool;
  isfriend: NumericBool;
  isfamily: NumericBool;
}

export interface PhotosResponse {
  page: number;
  pages: number;
  perpage: number;
  total: number;
  photo: Photo[];
}

export enum PhotoSize {
  thumbnail75 = 's',
  thumbnail100 = 't',
  thumbnail150 = 'q',
  small240 = 'm',
  small320 = 'n',
  small400 = 'w',
  medium800 = 'c',
  large1024 = 'b',
  large1600 = 'h',
  large2048 = 'k',
}

export interface SearchState {
  query: string;
  fetching: boolean;
  loadingMore: boolean;
  data: Photo[];
  error: Error | null;
  totalPages: number;
}

interface SearchStartAction {
  type: typeof SEARCH_START;
  name: string;
}

interface SearchSuccessAction {
  type: typeof SEARCH_SUCCESS;
  totalPages: number;
  photos: Photo[];
}

interface SearchFailureAction {
  type: typeof SEARCH_FAILURE;
  error: Error;
}

interface LoadMoreStartAction {
  type: typeof LOAD_MORE_START;
}

interface LoadMoreSuccessAction {
  type: typeof LOAD_MORE_SUCCESS;
  photos: Photo[];
}

interface LoadMoreFailureAction {
  type: typeof LOAD_MORE_FAILURE;
  error: Error;
}

interface TrackOpenURL {
  type: typeof TRACK_OPEN_URL;
}

export type Action =
  | SearchStartAction
  | SearchSuccessAction
  | SearchFailureAction
  | LoadMoreStartAction
  | LoadMoreSuccessAction
  | LoadMoreFailureAction
  | TrackOpenURL;
