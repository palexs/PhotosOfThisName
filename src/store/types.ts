import {ThunkDispatch} from 'redux-thunk';
import {Linking, PermissionsAndroid, Platform} from 'react-native';
import Contacts from 'react-native-contacts';
import store from './store';
import {Action as SearchAction} from './photos/types';
import {Action as ContactAction} from './contacts/types';

export interface Dependencies {
  Linking: typeof Linking;
  Contacts: typeof Contacts;
  PermissionsAndroid: typeof PermissionsAndroid;
  Platform: typeof Platform;
}

export type RootState = ReturnType<typeof store.getState>;
export type GetRootState = () => RootState;

type Action = SearchAction | ContactAction;

export type AppDispatch = typeof store.dispatch &
  ThunkDispatch<RootState, Dependencies, Action>;
