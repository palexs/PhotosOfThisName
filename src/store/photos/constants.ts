export const SEARCH_START = 'PHOTOS/SEARCH_START';
export const SEARCH_SUCCESS = 'PHOTOS/SEARCH_SUCCESS';
export const SEARCH_FAILURE = 'PHOTOS/SEARCH_FAILURE';
export const LOAD_MORE_START = 'PHOTOS/LOAD_MORE_START';
export const LOAD_MORE_SUCCESS = 'PHOTOS/LOAD_MORE_SUCCESS';
export const LOAD_MORE_FAILURE = 'PHOTOS/LOAD_MORE_FAILURE';
export const GET_LOCATION_START = 'PHOTOS/GET_LOCATION_START';
export const GET_LOCATION_SUCCESS = 'PHOTOS/GET_LOCATION_SUCCESS';
export const GET_LOCATION_FAILURE = 'PHOTOS/GET_LOCATION_FAILURE';
export const TRACK_OPEN_URL = 'PHOTOS/TRACK_OPEN_URL';
