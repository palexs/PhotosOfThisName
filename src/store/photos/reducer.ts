import {Action, PhotosState} from './types';
import {
  SEARCH_START,
  SEARCH_SUCCESS,
  SEARCH_FAILURE,
  LOAD_MORE_START,
  LOAD_MORE_SUCCESS,
  LOAD_MORE_FAILURE,
  GET_LOCATION_SUCCESS,
  GET_LOCATION_FAILURE,
  GET_LOCATION_START,
} from './constants';

const initialState: PhotosState = {
  query: '',
  fetching: false,
  loadingMore: false,
  data: [],
  error: null,
  totalPages: 0,
  locations: {},
};

const reducer = (state: PhotosState = initialState, action: Action) => {
  switch (action.type) {
    case SEARCH_START: {
      const {name} = action;
      return {
        ...state,
        query: name,
        fetching: true,
        data: [],
        totalPages: 0,
        error: null,
        locations: {},
      };
    }
    case SEARCH_SUCCESS: {
      const {photos, totalPages} = action;
      return {
        ...state,
        data: photos,
        totalPages,
        fetching: false,
      };
    }
    case SEARCH_FAILURE: {
      const {error} = action;
      return {
        ...state,
        fetching: false,
        data: [],
        locations: {},
        error,
      };
    }
    case LOAD_MORE_START: {
      return {
        ...state,
        loadingMore: true,
        error: null,
      };
    }
    case LOAD_MORE_SUCCESS: {
      const {photos} = action;
      return {
        ...state,
        loadingMore: false,
        data: [...state.data, ...photos],
      };
    }
    case LOAD_MORE_FAILURE: {
      const {error} = action;
      return {
        ...state,
        loadingMore: false,
        data: [],
        error,
      };
    }
    case GET_LOCATION_START: {
      const {photoID} = action;
      return {
        ...state,
        locations: {
          ...state.locations,
          [photoID]: {
            loading: true,
            error: null,
            country: null,
          },
        },
      };
    }
    case GET_LOCATION_SUCCESS: {
      const {photoID, country} = action;
      return {
        ...state,
        locations: {
          ...state.locations,
          [photoID]: {
            loading: false,
            country,
            error: null,
          },
        },
      };
    }
    case GET_LOCATION_FAILURE: {
      const {photoID, error} = action;
      return {
        ...state,
        locations: {
          ...state.locations,
          [photoID]: {
            loading: false,
            country: null,
            error,
          },
        },
      };
    }
    default:
      return state;
  }
};

export default reducer;
