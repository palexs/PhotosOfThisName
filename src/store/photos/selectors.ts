import {Photo, PhotoSize, PhotoLocation} from './types';
import {RootState} from '../types';

export const getQuery = (state: RootState) => state.photos.query;

export const getFetching = (state: RootState) => state.photos.fetching;

export const getData = (state: RootState) => state.photos.data;

export const getTotalPages = (state: RootState) => state.photos.totalPages;

export const getLoadingMore = (state: RootState) => state.photos.loadingMore;

export const getError = (state: RootState) => state.photos.error;

export const getPhotoLocation = (state: RootState, photoID: string) =>
  (state.photos.locations as {[photoID: string]: PhotoLocation})[photoID];

export const getSizedImageUrlForPhoto = (
  photo: Photo,
  size: PhotoSize = PhotoSize.small240,
) => {
  return `https://live.staticflickr.com/${photo.server}/${photo.id}_${photo.secret}_${size}.jpg`;
};
