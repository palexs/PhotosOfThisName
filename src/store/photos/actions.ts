import {ActionCreator, Dispatch} from 'redux';
import {fetchPhotos, fetchPhotoLocation} from '../../FlickrAPI';
import {AppDispatch, GetRootState, Dependencies} from '../types';
import {
  SEARCH_START,
  SEARCH_SUCCESS,
  SEARCH_FAILURE,
  LOAD_MORE_START,
  LOAD_MORE_SUCCESS,
  LOAD_MORE_FAILURE,
  TRACK_OPEN_URL,
  GET_LOCATION_START,
  GET_LOCATION_SUCCESS,
  GET_LOCATION_FAILURE,
  TRACK_OPEN_URL_FAILURE,
} from './constants';
import {
  Action,
  GetLocationFailure,
  GetLocationStart,
  GetLocationSuccess,
  LoadMoreFailure,
  LoadMoreStart,
  LoadMoreSuccess,
  PhotoLocationResponse,
  PhotosResponse,
  SearchFailure,
  SearchStart,
  SearchSuccess,
  ThunkActionCreator,
  TrackOpenURL,
  TrackOpenURLFailure,
} from './types';
import {getQuery} from './selectors';

export const searchStart: ActionCreator<SearchStart> = (name: string) => ({
  type: SEARCH_START,
  name,
});

export const searchSuccess: ActionCreator<SearchSuccess> = (
  response: PhotosResponse,
) => ({
  type: SEARCH_SUCCESS,
  totalPages: response.pages,
  photos: response.photo,
});

export const searchFailure: ActionCreator<SearchFailure> = (error: Error) => ({
  type: SEARCH_FAILURE,
  error,
});

export const loadMoreStart: ActionCreator<LoadMoreStart> = () => ({
  type: LOAD_MORE_START,
});

export const loadMoreSuccess: ActionCreator<LoadMoreSuccess> = (
  response: PhotosResponse,
) => ({type: LOAD_MORE_SUCCESS, photos: response.photo});

export const loadMoreFailure: ActionCreator<LoadMoreFailure> = (
  error: Error,
) => ({
  type: LOAD_MORE_FAILURE,
  error,
});

export const getLocationStart: ActionCreator<GetLocationStart> = (
  photoID: string,
) => ({
  type: GET_LOCATION_START,
  photoID,
});

export const getLocationSuccess: ActionCreator<GetLocationSuccess> = (
  response: PhotoLocationResponse,
  photoID: string,
) => ({
  type: GET_LOCATION_SUCCESS,
  country: response.location.country._content,
  photoID,
});

export const getLocationError: ActionCreator<GetLocationFailure> = (
  error: Error,
  photoID: string,
) => ({
  type: GET_LOCATION_FAILURE,
  error,
  photoID,
});

export const trackOpenURL: ActionCreator<TrackOpenURL> = (url: string) => ({
  type: TRACK_OPEN_URL,
  url,
});

export const trackOpenURLFailure: ActionCreator<TrackOpenURLFailure> = (
  error: Error,
) => ({
  type: TRACK_OPEN_URL_FAILURE,
  error,
});

export const search: ThunkActionCreator =
  (name: string) => async (dispatch: Dispatch<Action>) => {
    dispatch(searchStart(name));
    try {
      const response = await fetchPhotos(name, 1);
      dispatch(searchSuccess(response));
    } catch (error) {
      dispatch(searchFailure(error));
    }
  };

export const getLocation: ThunkActionCreator =
  (photoID: string) => async (dispatch: Dispatch<Action>) => {
    dispatch(getLocationStart(photoID));
    try {
      const response = await fetchPhotoLocation(photoID);
      dispatch(getLocationSuccess(response, photoID));
    } catch (error) {
      dispatch(getLocationError(error, photoID));
    }
  };

export const loadMore: ThunkActionCreator =
  (page: number) => async (dispatch: AppDispatch, getState: GetRootState) => {
    dispatch(loadMoreStart());
    try {
      const state = getState();
      const response = await fetchPhotos(getQuery(state), page);
      dispatch(loadMoreSuccess(response));
    } catch (error) {
      dispatch(loadMoreFailure(error));
    }
  };

export const openURL: ThunkActionCreator =
  (url: string) =>
  async (
    dispatch: AppDispatch,
    getState: GetRootState,
    {Linking, Alert}: Dependencies,
  ) => {
    const isSupported = await Linking.canOpenURL(url);
    if (isSupported) {
      try {
        await Linking.openURL('url');
        dispatch(trackOpenURL(url));
      } catch (error) {
        dispatch(trackOpenURLFailure(error));
      }
    } else {
      Alert.alert('URI not supported', `Can't open URI: ${url}`, [
        {
          text: 'Ok',
          style: 'cancel',
        },
      ]);
    }
  };
