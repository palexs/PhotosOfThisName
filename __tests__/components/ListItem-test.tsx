import React from 'react';
import renderer from 'react-test-renderer';
import PhotoItem from '../../src/components/ListItem';
import {Photo} from '../../src/store/photos/types';
import getMockStore from '../helpers';
import {Provider} from 'react-redux';

describe('ListItem component', () => {
  const photo: Photo = {
    id: '12345',
    owner: 'owner1',
    secret: 'secret1',
    server: 'server1',
    farm: 1,
    title: 'title1',
    ispublic: 1,
    isfriend: 0,
    isfamily: 0,
  };

  it('renders - no country info', () => {
    expect.assertions(1);
    const store = getMockStore({
      state: {
        photos: {
          locations: {},
        },
      },
    });

    const component = renderer
      .create(
        <Provider store={store}>
          <PhotoItem item={photo} onItemPress={jest.fn()} />
        </Provider>,
      )
      .toJSON();
    expect(component).toMatchSnapshot();
  });

  it('renders - has country info', () => {
    expect.assertions(1);
    const store = getMockStore({
      state: {
        photos: {
          locations: {
            '12345': {
              loading: false,
              country: 'Ukraine',
              error: null,
            },
          },
        },
      },
    });

    const component = renderer
      .create(
        <Provider store={store}>
          <PhotoItem item={photo} onItemPress={jest.fn()} />
        </Provider>,
      )
      .toJSON();
    expect(component).toMatchSnapshot();
  });
});
