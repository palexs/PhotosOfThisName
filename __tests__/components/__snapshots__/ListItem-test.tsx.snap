// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem component renders - has country info 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "marginHorizontal": 10,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#ffffff",
        "borderBottomWidth": 0.5,
        "borderColor": "#bcbbc1",
        "flexDirection": "row",
        "padding": 14,
      }
    }
    testID="padView"
  >
    <View
      accessibilityIgnoresInvertColors={true}
      style={
        Object {
          "backgroundColor": "transparent",
          "overflow": "hidden",
          "position": "relative",
        }
      }
    >
      <Image
        onLoad={[Function]}
        source={
          Object {
            "uri": "https://live.staticflickr.com/server1/12345_secret1_m.jpg",
          }
        }
        style={
          Object {
            "bottom": 0,
            "height": 60,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "width": 60,
          }
        }
        testID="RNE__Image"
        theme={
          Object {
            "colors": Object {
              "black": "#242424",
              "disabled": "hsl(208, 8%, 90%)",
              "divider": "#bcbbc1",
              "error": "#ff190c",
              "grey0": "#393e42",
              "grey1": "#43484d",
              "grey2": "#5e6977",
              "grey3": "#86939e",
              "grey4": "#bdc6cf",
              "grey5": "#e1e8ee",
              "greyOutline": "#bbb",
              "platform": Object {
                "android": Object {
                  "error": "#f44336",
                  "grey": "rgba(0, 0, 0, 0.54)",
                  "primary": "#2196f3",
                  "searchBg": "#dcdce1",
                  "secondary": "#9C27B0",
                  "success": "#4caf50",
                  "warning": "#ffeb3b",
                },
                "default": Object {
                  "error": "#ff3b30",
                  "grey": "#7d7d7d",
                  "primary": "#007aff",
                  "searchBg": "#dcdce1",
                  "secondary": "#5856d6",
                  "success": "#4cd964",
                  "warning": "#ffcc00",
                },
                "ios": Object {
                  "error": "#ff3b30",
                  "grey": "#7d7d7d",
                  "primary": "#007aff",
                  "searchBg": "#dcdce1",
                  "secondary": "#5856d6",
                  "success": "#4cd964",
                  "warning": "#ffcc00",
                },
                "web": Object {
                  "error": "#ff190c",
                  "grey": "#393e42",
                  "primary": "#2089dc",
                  "searchBg": "#303337",
                  "secondary": "#ca71eb",
                  "success": "#52c41a",
                  "warning": "#faad14",
                },
              },
              "primary": "#2089dc",
              "searchBg": "#303337",
              "secondary": "#ca71eb",
              "success": "#52c41a",
              "warning": "#faad14",
              "white": "#ffffff",
            },
          }
        }
        transition={true}
        transitionDuration={360}
      />
      <View
        accessibilityElementsHidden={true}
        importantForAccessibility="no-hide-descendants"
        pointerEvents="none"
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "opacity": 1,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#bdbdbd",
              "height": 60,
              "justifyContent": "center",
              "width": 60,
            }
          }
          testID="RNE__Image__placeholder"
        />
      </View>
      <View
        style={
          Object {
            "height": 60,
            "width": 60,
          }
        }
        testID="RNE__Image__children__container"
      />
    </View>
    <View
      style={
        Object {
          "paddingLeft": 16,
        }
      }
    />
    <View
      style={
        Array [
          Object {
            "alignItems": "flex-start",
            "flex": 1,
            "justifyContent": "center",
          },
          undefined,
        ]
      }
      theme={
        Object {
          "colors": Object {
            "black": "#242424",
            "disabled": "hsl(208, 8%, 90%)",
            "divider": "#bcbbc1",
            "error": "#ff190c",
            "grey0": "#393e42",
            "grey1": "#43484d",
            "grey2": "#5e6977",
            "grey3": "#86939e",
            "grey4": "#bdc6cf",
            "grey5": "#e1e8ee",
            "greyOutline": "#bbb",
            "platform": Object {
              "android": Object {
                "error": "#f44336",
                "grey": "rgba(0, 0, 0, 0.54)",
                "primary": "#2196f3",
                "searchBg": "#dcdce1",
                "secondary": "#9C27B0",
                "success": "#4caf50",
                "warning": "#ffeb3b",
              },
              "default": Object {
                "error": "#ff3b30",
                "grey": "#7d7d7d",
                "primary": "#007aff",
                "searchBg": "#dcdce1",
                "secondary": "#5856d6",
                "success": "#4cd964",
                "warning": "#ffcc00",
              },
              "ios": Object {
                "error": "#ff3b30",
                "grey": "#7d7d7d",
                "primary": "#007aff",
                "searchBg": "#dcdce1",
                "secondary": "#5856d6",
                "success": "#4cd964",
                "warning": "#ffcc00",
              },
              "web": Object {
                "error": "#ff190c",
                "grey": "#393e42",
                "primary": "#2089dc",
                "searchBg": "#303337",
                "secondary": "#ca71eb",
                "success": "#52c41a",
                "warning": "#faad14",
              },
            },
            "primary": "#2089dc",
            "searchBg": "#303337",
            "secondary": "#ca71eb",
            "success": "#52c41a",
            "warning": "#faad14",
            "white": "#ffffff",
          },
        }
      }
    >
      <Text
        style={
          Object {
            "backgroundColor": "transparent",
            "color": "#242424",
            "fontSize": 17,
          }
        }
        testID="listItemTitle"
      >
        title1
      </Text>
      <View
        style={
          Object {
            "flexDirection": "row",
            "height": 20,
          }
        }
      >
        <View
          style={
            Object {
              "overflow": "hidden",
            }
          }
        >
          <View
            style={
              Object {
                "paddingRight": 5,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "justifyContent": "center",
                }
              }
            >
              <Text
                allowFontScaling={false}
                style={
                  Array [
                    Object {
                      "color": "black",
                      "fontSize": 16,
                    },
                    Object {
                      "backgroundColor": "transparent",
                    },
                    Object {
                      "fontFamily": "Ionicons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
                testID="iconIcon"
              >
                
              </Text>
            </View>
          </View>
        </View>
        <Text
          style={
            Object {
              "backgroundColor": "transparent",
              "color": "#242424",
              "fontSize": 15,
            }
          }
          testID="listItemTitle"
        >
          owner1
        </Text>
      </View>
      <View
        style={
          Object {
            "flexDirection": "row",
            "height": 20,
          }
        }
      >
        <View
          style={
            Object {
              "overflow": "hidden",
            }
          }
        >
          <View
            style={
              Object {
                "paddingRight": 5,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "justifyContent": "center",
                }
              }
            >
              <Text
                allowFontScaling={false}
                style={
                  Array [
                    Object {
                      "color": "black",
                      "fontSize": 16,
                    },
                    Object {
                      "backgroundColor": "transparent",
                    },
                    Object {
                      "fontFamily": "Ionicons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
                testID="iconIcon"
              >
                
              </Text>
            </View>
          </View>
        </View>
        <Text
          style={
            Object {
              "backgroundColor": "transparent",
              "color": "#242424",
              "fontSize": 15,
            }
          }
          testID="listItemTitle"
        >
          Ukraine
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`ListItem component renders - no country info 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "marginHorizontal": 10,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#ffffff",
        "borderBottomWidth": 0.5,
        "borderColor": "#bcbbc1",
        "flexDirection": "row",
        "padding": 14,
      }
    }
    testID="padView"
  >
    <View
      accessibilityIgnoresInvertColors={true}
      style={
        Object {
          "backgroundColor": "transparent",
          "overflow": "hidden",
          "position": "relative",
        }
      }
    >
      <Image
        onLoad={[Function]}
        source={
          Object {
            "uri": "https://live.staticflickr.com/server1/12345_secret1_m.jpg",
          }
        }
        style={
          Object {
            "bottom": 0,
            "height": 60,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "width": 60,
          }
        }
        testID="RNE__Image"
        theme={
          Object {
            "colors": Object {
              "black": "#242424",
              "disabled": "hsl(208, 8%, 90%)",
              "divider": "#bcbbc1",
              "error": "#ff190c",
              "grey0": "#393e42",
              "grey1": "#43484d",
              "grey2": "#5e6977",
              "grey3": "#86939e",
              "grey4": "#bdc6cf",
              "grey5": "#e1e8ee",
              "greyOutline": "#bbb",
              "platform": Object {
                "android": Object {
                  "error": "#f44336",
                  "grey": "rgba(0, 0, 0, 0.54)",
                  "primary": "#2196f3",
                  "searchBg": "#dcdce1",
                  "secondary": "#9C27B0",
                  "success": "#4caf50",
                  "warning": "#ffeb3b",
                },
                "default": Object {
                  "error": "#ff3b30",
                  "grey": "#7d7d7d",
                  "primary": "#007aff",
                  "searchBg": "#dcdce1",
                  "secondary": "#5856d6",
                  "success": "#4cd964",
                  "warning": "#ffcc00",
                },
                "ios": Object {
                  "error": "#ff3b30",
                  "grey": "#7d7d7d",
                  "primary": "#007aff",
                  "searchBg": "#dcdce1",
                  "secondary": "#5856d6",
                  "success": "#4cd964",
                  "warning": "#ffcc00",
                },
                "web": Object {
                  "error": "#ff190c",
                  "grey": "#393e42",
                  "primary": "#2089dc",
                  "searchBg": "#303337",
                  "secondary": "#ca71eb",
                  "success": "#52c41a",
                  "warning": "#faad14",
                },
              },
              "primary": "#2089dc",
              "searchBg": "#303337",
              "secondary": "#ca71eb",
              "success": "#52c41a",
              "warning": "#faad14",
              "white": "#ffffff",
            },
          }
        }
        transition={true}
        transitionDuration={360}
      />
      <View
        accessibilityElementsHidden={true}
        importantForAccessibility="no-hide-descendants"
        pointerEvents="none"
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "opacity": 1,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#bdbdbd",
              "height": 60,
              "justifyContent": "center",
              "width": 60,
            }
          }
          testID="RNE__Image__placeholder"
        />
      </View>
      <View
        style={
          Object {
            "height": 60,
            "width": 60,
          }
        }
        testID="RNE__Image__children__container"
      />
    </View>
    <View
      style={
        Object {
          "paddingLeft": 16,
        }
      }
    />
    <View
      style={
        Array [
          Object {
            "alignItems": "flex-start",
            "flex": 1,
            "justifyContent": "center",
          },
          undefined,
        ]
      }
      theme={
        Object {
          "colors": Object {
            "black": "#242424",
            "disabled": "hsl(208, 8%, 90%)",
            "divider": "#bcbbc1",
            "error": "#ff190c",
            "grey0": "#393e42",
            "grey1": "#43484d",
            "grey2": "#5e6977",
            "grey3": "#86939e",
            "grey4": "#bdc6cf",
            "grey5": "#e1e8ee",
            "greyOutline": "#bbb",
            "platform": Object {
              "android": Object {
                "error": "#f44336",
                "grey": "rgba(0, 0, 0, 0.54)",
                "primary": "#2196f3",
                "searchBg": "#dcdce1",
                "secondary": "#9C27B0",
                "success": "#4caf50",
                "warning": "#ffeb3b",
              },
              "default": Object {
                "error": "#ff3b30",
                "grey": "#7d7d7d",
                "primary": "#007aff",
                "searchBg": "#dcdce1",
                "secondary": "#5856d6",
                "success": "#4cd964",
                "warning": "#ffcc00",
              },
              "ios": Object {
                "error": "#ff3b30",
                "grey": "#7d7d7d",
                "primary": "#007aff",
                "searchBg": "#dcdce1",
                "secondary": "#5856d6",
                "success": "#4cd964",
                "warning": "#ffcc00",
              },
              "web": Object {
                "error": "#ff190c",
                "grey": "#393e42",
                "primary": "#2089dc",
                "searchBg": "#303337",
                "secondary": "#ca71eb",
                "success": "#52c41a",
                "warning": "#faad14",
              },
            },
            "primary": "#2089dc",
            "searchBg": "#303337",
            "secondary": "#ca71eb",
            "success": "#52c41a",
            "warning": "#faad14",
            "white": "#ffffff",
          },
        }
      }
    >
      <Text
        style={
          Object {
            "backgroundColor": "transparent",
            "color": "#242424",
            "fontSize": 17,
          }
        }
        testID="listItemTitle"
      >
        title1
      </Text>
      <View
        style={
          Object {
            "flexDirection": "row",
            "height": 20,
          }
        }
      >
        <View
          style={
            Object {
              "overflow": "hidden",
            }
          }
        >
          <View
            style={
              Object {
                "paddingRight": 5,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "justifyContent": "center",
                }
              }
            >
              <Text
                allowFontScaling={false}
                style={
                  Array [
                    Object {
                      "color": "black",
                      "fontSize": 16,
                    },
                    Object {
                      "backgroundColor": "transparent",
                    },
                    Object {
                      "fontFamily": "Ionicons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
                testID="iconIcon"
              >
                
              </Text>
            </View>
          </View>
        </View>
        <Text
          style={
            Object {
              "backgroundColor": "transparent",
              "color": "#242424",
              "fontSize": 15,
            }
          }
          testID="listItemTitle"
        >
          owner1
        </Text>
      </View>
      <View
        style={
          Object {
            "flexDirection": "row",
            "height": 20,
          }
        }
      />
    </View>
  </View>
</View>
`;
